docs_src/design/adr/core/0026-Core-Data-Cache.md

```diff
@@ -0,0 +1,85 @@
+# Core Data Cache
+
+## Status
+
+**Draft**
+
+proposed: 4/4/22
+
+## Context
+There is a desire to have core data serve more as a data cache - that is, keeping only the "latest" readings.  There is also a desire to allow for queries of core data to serve up the "latest" readings.  This ADR defines new operations and APIs to satisfy these needs.
+
+### Queries to return "Latest"
+
+Core data will add the following new operations (and externally visible APIs) to provide the means to get the `latest` readings.
+
+- Query for the latest N readings by device name
+    - API: *GET* /reading/device/name/{name}/latest/{N}
+    - returns the last N readings for the specified device
+    - N must be > 0
+- Query for the latest N readings by device name and device resource
+    - API: *GET* /reading/device/name/{name}/resourceName/{resourceName}/latest/{N}
+    - returns the latest N readings for the specified device and specified device resource
+    - N must be > 0
```

Comment:
Instead of adding these new APIs, we can use the query parameters (`offset` and `limit`) with the existing API to achieve the purpose
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.1.0#/default/get_reading_device_name__deviceName__resourceName__resourceName_
For example, `*GET* /reading/device/name/{name}/resourceName/{resourceName}?limit=N` can get the latest N Reading for the specified device and specified device resource
Also, this usage can work for all the EdgeX GET API with multiple return values.

Reply:
I agree that we could use the limit query parameter to specify N for both APIs.
In that way, the existing  API: *GET* /reading/device/name/{name} is unchanged. It is still not very useful to us in this form (we will get N readings of the chattiest resources, and still have to use the second API to get the others).
The second API is new, however; there is no existing core-data API with required functionality. We could use it in the form that Cloud showed, with the ?limit=N query parameter; that will be more consistent with the existing APIs.
Don't know that we need to support the "offset=" query parameter for it; I don't see us using it that way.

Reply:
@TomBrennan-Eaton the second API is
GET /reading/device/name/{deviceName}/resourceName/{resourceName}
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.1.0#/default/get_reading_device_name__deviceName__resourceName__resourceName_

Reply:
Thanks for pointing that out, Cloud. This API is new in 2.1, and I had been only looking at 2.0, where it doesn't exist.
Yes, this is what we were after here (I would want to use the version of the API that doesn't have start and end times).